### PR TITLE
fix: the input field of tool panel not worked as expected

### LINKED ISSUE #6002: the input field of tool panel not worked as expected
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.6.12-fix1

### Cloud or Self Hosted

Self Hosted (Source)

### Steps to reproduce

when you develop a tool, add this config in the yaml file:
```
parameters:
  - name: test_select
    type: select
    options:
      - value: 0
        label:
          en_US: value_is_0
    required: true
    label:
      en_US: Test Select
    human_description:
      en_US: Test Select
    form: form
  - name: test_secret_input
    type: secret-input
    required: true
    label:
      en_US: Test Secret Input
    human_description:
      en_US: Test Secret Input
    form: form
  - name: test_number
    type: number
    required: true
    label:
      en_US: Test Number
    human_description:
      en_US: Test Number
    form: form
  - name: test_boolean
    type: boolean
    required: true
    label:
      en_US: Test Boolean
    human_description:
      en_US: Test Boolean
    form: form
```

you will get some input fields like this:
image

the problem is:

- secret input field,  it is not mask the input text
- number input field, you can't enter a `0`, it will not display; if you enter the `0.23`, it will display `.23`
- number input field, if the default value is 0,  when you run this node, it will raise `tool parameter test_number not found in tool config`
- select field, when you choose an option run once, and then click the clear btn run once again, it will submit the first time's value
- boolean field, it is not display the required star
- boolean field, there is no default value, but the `False` is selected in the page. when you click run btn, it will raise `Test Boolean 不能为空`
- boolean field, when you click again to choose `False`, it will raise `tool parameter test_boolean not found in tool config`

### ✔️ Expected Behavior

_No response_

### ❌ Actual Behavior

_No response_

### PR DESCRIPTION
**Pull request by hjlarry**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

Fixes https://github.com/langgenius/dify/issues/6002

## Type of Change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [ ] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/core/tools/tool_manager.py [hunk 1]
```diff
@@ -201,7 +201,7 @@ def _init_runtime_parameter(cls, parameter_rule: ToolParameter, parameters: dict
             init runtime parameter
         """
         parameter_value = parameters.get(parameter_rule.name)
-        if not parameter_value:
+        if not parameter_value and parameter_value != 0:
```
Comment (Yeuoly) [inline api/core/tools/tool_manager.py:204]: what about change it to `if parameter_value == None` ?

Comment (hjlarry) [reply api/core/tools/tool_manager.py:204]: actually, I can't reproduce all the false value of parameter_value here, cause of the frontend also has validation.
I see the `ToolParameterType` has string,boolean,select,file,  maybe it is an empty string or None here?  so I only add `!=0` here.

Comment (Yeuoly) [reply api/core/tools/tool_manager.py:204]: Got it, frontend has already ensured all parameters are in correct type, validation here need only to check if it's empty, SGTM

**Review by Yeuoly — APPROVED** [review]
LGTM

**Review by crazywoola — APPROVED** [review]

**Pull request merged**